rllib/env/tests/test_policy_client_server_setup.sh

```diff
@@ -85,4 +85,53 @@ client2_pid=$!
 sleep 2
 python $basedir/$client_script --inference-mode=$inference_mode --port=$worker_2_port "$stop_criterion"
 
+client_1_is_running=true
+client_2_is_running=true
+server_is_running=true
+
+if ! ps -p $client1_pid > /dev/null
+then
+   echo "$client1_pid client1_pid is not running"
+    client_1_is_running=false
+fi
+
+if ! ps -p $client2_pid > /dev/null
+then
+   echo "$client2_pid client2_pid is not running"
+    client_2_is_running=false
+fi
+
+if ! ps -p $server_pid > /dev/null
+then
+   echo "$server_pid server_pid is not running"
+    server_is_running=false
+fi
+
+if ! $client_1_is_running; then
+  wait $client1_pid
+  client1_exit_code=$?
+  if [ $client1_exit_code -ne 0 ]; then
+    echo "Client 1 exited with code $client1_exit_code"
+    exit $client1_exit_code
+  fi
+fi
+
+if ! $client_2_is_running; then
+  wait $client2_pid
+  client2_exit_code=$?
+  if [ $client2_exit_code -ne 0 ]; then
+    echo "Client 2 exited with code $client2_exit_code"
+    exit $client2_exit_code
+  fi
+fi
+
+if ! $server_is_running; then
+  wait $server_pid
+  server_exit_code=$?
+  if [ $server_exit_code -ne 0 ]; then
+    echo "Server exited with code $server_exit_code"
+    exit $server_exit_code
+  fi
+fi
```

Comment:
can you make a function for checking and potentially exiting, like:
```
exit_if_not_running()
{
    local pid=$1

    if ps -p $pid> /dev/null
    then
        return
    fi

    echo "$2 is not running"

    wait $pid
    exit_code=$?
    exit $exit_code
}

maybe_exit $client1_pid "client 1"
maybe_exit $client2_pid "client 2"
maybe_exit $server_pid "server"
```

Reply:
sure! to be honest I only didn't before because I don't know how to write functions in bash

Reply:
done